ngraph/core/reference/include/ngraph/runtime/reference/deformable_convolution.hpp

```diff
@@ -76,20 +84,37 @@ namespace ngraph
                                                     const float x_idx,
                                                     const float y_idx,
                                                     const int x_size,
-                                                    const int y_size)
+                                                    const int y_size,
+                                                    const bool use_pad)
                 {
-                    const int x1 = std::max(static_cast<int>(std::floor(x_idx)), 0);
-                    const int x2 = std::min(static_cast<int>(std::ceil(x_idx)), x_size - 1);
-                    const int y1 = std::max(static_cast<int>(std::floor(y_idx)), 0);
-                    const int y2 = std::min(static_cast<int>(std::ceil(y_idx)), y_size - 1);
+                    const int y1 = use_pad ? static_cast<int>(std::floor(y_idx))
+                                           : std::max(static_cast<int>(std::floor(y_idx)), 0);
+                    const int x1 = use_pad ? static_cast<int>(std::floor(x_idx))
+                                           : std::max(static_cast<int>(std::floor(x_idx)), 0);
+
+                    const int y2 =
+                        use_pad ? y1 + 1 : std::min(static_cast<int>(std::ceil(y_idx)), y_size - 1);
+                    const int x2 =
+                        use_pad ? x1 + 1 : std::min(static_cast<int>(std::ceil(x_idx)), x_size - 1);
 
                     const float distX = x_idx - x1;
                     const float distY = y_idx - y1;
 
-                    const float value11 = data[y1 * x_size + x1];
-                    const float value12 = data[y2 * x_size + x1];
-                    const float value21 = data[y1 * x_size + x2];
-                    const float value22 = data[y2 * x_size + x2];
+                    float value11 = 0;
```

Comment:
should it be something like data[0] instead of 0? Or maybe something even more complicated...

Reply:
no, it shouldn't, see paddlepaddle https://github.com/PaddlePaddle/Paddle/blob/bd2a4e23fad95fbd10f5ea33f1ab922b65be6515/paddle/fluid/operators/deformable_conv_func.h#L131 and mxnet https://github.com/apache/incubator-mxnet/blob/77670c54d7ad5a9bb97fc3dd916a9eecdc4f43c1/src/operator/contrib/nn/modulated_deformable_im2col.h#L105 implementations